nixos/modules/services/networking/wpa_supplicant.nix

```diff
@@ -68,14 +74,27 @@ let
       wantedBy = [ "multi-user.target" ];
       stopIfChanged = false;
 
-      path = [ package ];
+      path = [ package pkgs.gawk ];
+      serviceConfig.RuntimeDirectory = "wpa_supplicant";
+      serviceConfig.EnvironmentFile = mkIf (cfg.environmentFile != null)
+        (builtins.toPath cfg.environmentFile);
 
       script =
       ''
-        if [ -f /etc/wpa_supplicant.conf -a "/etc/wpa_supplicant.conf" != "${configFile}" ]; then
+        if [ -f /etc/wpa_supplicant.conf -a "${toString configIsGenerated}" = 1 ]; then
           echo >&2 "<3>/etc/wpa_supplicant.conf present but ignored. Generated ${configFile} is used instead."
         fi
```

Comment:
It's probably better to wrap the `if` in an `optionalString configIsGenerated`.

Reply:
Done.